src/libcore/slice/mod.rs

```diff
@@ -4152,6 +4152,23 @@ impl<'a, T> DoubleEndedIterator for Chunks<'a, T> {
             Some(snd)
         }
     }
+
+    #[inline]
+    fn nth_back(&mut self, n: usize) -> Option<Self::Item> {
+        let (end, overflow) = self.v.len().overflowing_sub(n * self.chunk_size);
+        if overflow {
+            self.v = &mut [];
+            None
+        } else {
+            let start = match end.checked_sub(self.chunk_size) {
+                Some(res) => cmp::min(self.v.len(), res),
```

Comment:
I think this match arm can just return `res`? `res` is guaranteed to be smaller than `self.v.len()` here. 🙂

Reply:
Hey @timvermeulen, thank you for reviewing my PR. And, yes that is true... I'll make the change and commit once all below comments are resolved.